### PR TITLE
Add `provider` API 

### LINKED ISSUE #6: Choose backend provider
It would be great to optionally choose the backend provider to query, and/or to specify the order of providers to query.

### PR DESCRIPTION
**Pull request by krishnagopinath**
Closes #6.

The API is quite simple!

```
isbn.provider([..]).resolve(..)
```

### CODE REVIEW

**Comment by krishnagopinath** [comment]
@palmerabollo -- It would be awesome if you took a look at this! 👀 

I also upgraded everything to minimally use ES2015 and add some Node.js restrictions to the README. 

If you think I shouldn't have made those changes, I'd be happy to remove them, because each change is in own commit! 

I also ran the test suite, added extra tests to test the `provider` API and everything worked! 🎉

File: index.js [hunk 1]
```diff
@@ -159,60 +177,123 @@ function _resolveWorldcat(isbn, options, callback) {
...
+   * 
+   * ```
+   * isbn
+   *  .provider([isbn.PROVIDER_NAMES.OPENLIBRARY, isbn.PROVIDER_NAMES.GOOGLE])
+   *  .resolve(...)
+   * ```
+   */
+  provider(providers) {
+    const providerValid = Array.isArray(providers);
+    if(!providerValid) throw new Error('`providers` must be an array.');
```
Comment (palmerabollo) [inline index.js]:
```suggestion
    if (!providerValid) throw new Error('`providers` must be an array.');
```

File: index.js [hunk 2]
```diff
@@ -159,60 +177,123 @@ function _resolveWorldcat(isbn, options, callback) {
...
+    if (!providers.length) return this;
+
+    // remove duplicates, if any
+    providers = [...new Set(providers)];
+  
+    // Check to see if there are any unsupported providers in the list.
+    const providersSupported = providers.reduce((acc, p) => {
+      return acc && DEFAULT_PROVIDERS.includes(p);
+    }, true);
+    if(!providersSupported) throw new Error('Please pass in supported providers.');
```
Comment (palmerabollo) [inline index.js]:
```suggestion
    if (!providersSupported) throw new Error('Please pass in supported providers.');
```

**Review by palmerabollo — APPROVED** [review]
I've added a couple of minor linting suggestions. Thanks for your first contribution, it looks really good to me.

File: index.js (hunk 2 above)
Comment (krishnagopinath) [reply index.js]: ✅

File: index.js (hunk 1 above)
Comment (krishnagopinath) [reply index.js]: ✅

**Comment by krishnagopinath** [comment]
Fixed them up, @palmerabollo!

**Comment by palmerabollo** [comment]
A new 1.5.0 release has just been published. Thank you again for your help.

**Comment by krishnagopinath** [comment]
🎉 Thank you for the quick release!

**Pull request merged**